sources/account_graph.move

```diff
@@ -76,13 +79,22 @@ module account_graph::account_graph {
         event::emit(GraphCreated{ graph_id });
     }
 
+    fun relationship_exists<AccountProps: copy + drop + store, RelationshipProps: copy + drop + store>(
+        self: &AccountGraph<AccountProps, RelationshipProps>,
+        source: address,
+        target: address,
+    ): bool {
+        table::contains(&self.relationships, source) && vec_set::contains(table::borrow(&self.relationships, source), &target)
+    }
+
     /// Add a relationship as edge, where source is the sender.
     public fun add_relationship<AccountProps: copy + drop + store, RelationshipProps: copy + drop + store>(
         self: &mut AccountGraph<AccountProps, RelationshipProps>,
         target: address,
         ctx: &mut TxContext,
     ) {
         let source = sender(ctx);
+        if (relationship_exists(self, source, target)) return;
```

Comment:
I don't think it's necessary to make `add` idempotent, because `remove` isn't. I think it's Ok to keep them that way.